src/service/meshService.js

```diff
@@ -0,0 +1,38 @@
+import log from "loglevel";
+import { waitThirtySeconds } from "./helper.js";
+
+export default class meshService {
+  constructor(loader,
+    senderService,
+    receiverService) {
+    this.loader = loader;
+    this.senderService = senderService;
+    this.receiverService = receiverService;
+
+    let logLevel = process.env.LOG_LEVEL || "DEBUG";
+    log.setLevel(log.levels[logLevel]);
+  }
+
+  async sendMessage() {
+    log.debug("\nsending a message");
+    await this.senderService.sendMessage();
+  }
+
+  async sendFile() {
+    log.debug("\nsending a file");
+    await this.senderService.createAndSendMessageChunks();
+  }
+
+  async receiveMessage(wait) {
+    if(wait === true){
+      log.debug("\nwaiting 30 seconds for mesh to process the message");
+      await waitThirtySeconds();
```

Comment:
you have given waitThirtySeconds the ability to override in its definition. Its only called once but doesn have the ability to override as you havent supplied params in receiveMessage that can be passed down to waitThirtySeconds

Reply:
If a value is provided then it will override the default 30 seconds. Andrew tested and found 30 seconds to be ideal for sending and receiving a file. This may be useful in the future incase we need a longer timeout.

Reply:
timeout is now an env variable